### PR TITLE
[MM-56004] Convert `./components/post_view/new_message_separator/new_message_separator.tsx`  from Class Component to Function Component

### PR DESCRIPTION
**Pull request by Syed-Ali-Abbas-Zaidi**
#### Summary

Convert `./components/post_view/new_message_separator/new_message_separator.tsx`  from Class Component to Function Component

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/25596
Jira https://mattermost.atlassian.net/browse/MM-56004

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by larkox — APPROVED** [review]
Great work!

File: webapp/channels/src/components/post_view/new_message_separator/new_message_separator.tsx [hunk 1]
```diff
@@ -17,38 +17,45 @@ type Props = {
...
+                    lastViewedAt={lastViewedAt}
+                    channelId={channelId}
+                    threadId={threadId}
+                />
+            );
+        });
+
+    return (
+        <div
+            ref={wrapperRef}
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/post_view/new_message_separator/new_message_separator.tsx:47]: i dont think ref can be passed this way, dont we have to use forwardRef? cc @larkox

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
Thanks @Syed-Ali-Abbas-Zaidi - looks good to me 👍 

cc// @M-ZubairAhmed

**Pull request merged**